src/chainrules.jl

```diff
@@ -12,9 +12,9 @@ ChainRulesCore.@scalar_rule(
     binomlogpdf(n::Real, p::Real, k::Real),
     @setup(z = digamma(n - k + 1)),
     (
-        digamma(n + 2) - z + log1p(-p) - 1 / (1 + n),
+        ChainRulesCore.NoTangent(),
```

Comment:
I have no idea what this does. Is it needed for this PR or could it be applied separately?

Reply:
It is required since otherwise tests fail: currently we test derivatives with respect to these arguments with a `BigFloat` hack - only the generic Julia implementation allows to perturb the integer-valued arguments but the R implementation is not defined (or rather returns -Inf) for these values. In this PR the Julia implementations are updated to match this behaviour (which e.g. is also expected by Distributions) and returns `-Inf` if `n` or `k` are no integers. So we can't test these derivatives anymore. And due to this definition of `binomlogpdf` it does not make sense to define them based on the continuous extension anymore (I think it's questionable if it ever did).